src/Api/API.php

```diff
@@ -738,41 +738,51 @@ protected function getItem($itemtype, $id, $params = [])
+            if ($itemtype == "Computer") {
+                $fields['_connections'] = [];
+                foreach ($CFG_GLPI["directconnect_types"] as $connect_type) {
+                    $connect_item = new $connect_type();
+                    if ($connect_item->canView()) {
+                        $connect_table = getTableForItemType($connect_type);
+                        $iterator = $DB->request([
+                            'SELECT'    => [
+                                'glpi_computers_items.id AS assoc_id',
+                                'glpi_computers_items.computers_id AS assoc_computers_id',
+                                'glpi_computers_items.itemtype AS assoc_itemtype',
+                                'glpi_computers_items.items_id AS assoc_items_id',
+                                'glpi_computers_items.is_dynamic AS assoc_is_dynamic',
+                                "$connect_table.*"
+                            ],
+                            'FROM'      => 'glpi_computers_items',
+                            'LEFT JOIN' => [
+                                $connect_table => [
+                                    'ON' => [
+                                        'glpi_computers_items'  => 'items_id',
+                                        $connect_table          => 'id'
+                                    ]
                                 ]
+                            ],
+                            'WHERE'     => [
+                                'computers_id'                      => $id,
+                                'itemtype'                          => $connect_type,
+                                'glpi_computers_items.is_deleted'   => 0
                             ]
-                        ],
-                        'WHERE'     => [
-                            'computers_id'                      => $id,
-                            'itemtype'                          => $connect_type,
-                            'glpi_computers_items.is_deleted'   => 0
-                        ]
-                    ]);
-                    foreach ($iterator as $data) {
-                        $fields['_connections'][$connect_type][] = $data;
+                        ]);
+                        foreach ($iterator as $data) {
+                            $fields['_connections'][$connect_type][] = $data;
+                        }
                     }
                 }
+            } else if ($itemtype == 'NetworkPort') {
+                $networkport_networkport = new \NetworkPort_NetworkPort();
+                $opposite_networkport_id = $networkport_networkport->getOppositeContact($id);
+                if ($opposite_networkport_id) {
+                    $fields['_connections']['NetworkPort'][] = [
+                        'id'       => $opposite_networkport_id,
+                        'assoc_id' => $networkport_networkport->fields['id']
+                    ];
+                }
```

Comment:
So wee need this
```suggestion
            } else if ($itemtype == 'NetworkPort') {
         $networkport_networkport = new \NetworkPort_NetworkPort();
         $opposite_networkport_id = $networkport_networkport->getOppositeContact($id);
         if ($opposite_networkport_id) {
            $fields['_connections']['NetworkPort'][] = [
               "assoc_networkports_id_2" => $opposite_networkport_id,
               "assoc_networkports_id_1" => $id,
               "assoc_id" => $networkport_networkport->fields['id']
            ]
         }
```

Reply:
Which key is the sought NetworkPort `id` in this output? Do you suggest a user compare his input id to both `assoc_networkports_id_1` and `assoc_networkports_id_2` to find the opposite port id?

Moreover the provided output doesn't reflect the true state of the DB: `$id` may be either `assoc_networkports_id_1` or `assoc_networkports_id_2`.

In summary, there is no `NetworkPort` in the output and sorry, but i this change makes no sense to me.